### PR TITLE
Add documentation about the BCrypt cost value in project settings

### PR DESCRIPTION
**Pull request by jveldhuizen**

### CODE REVIEW

File: content/refguide/project-settings.md [hunk 1]
```diff
@@ -103,15 +103,20 @@ The hash algorithm is used to generate hash values for attributes of the **Hashe
...
 
-#### 3.9.1 Performance
+#### 3.9.1 BCrypt Cost
 
-This performance difference is hardly noticeable to a single user when signing in (the password you enter when signing in is hashed using the selected algorithm), so in general the performance alone is not a reason to choose SSHA256 over BCrypt. This situation can change when dealing with high concurrency of hashing operations. A common example of an area where this occurs is published web services exposing operations that compute quickly, like short-running microflows.
+This value is used to specify the cost for the BCrypt algorithm. The cost value defines the number of rounds in which its underlying hash algorithm is applied. The number of rounds are calculated by using the cost value as power of 2. Mendix uses a default cost value of 10, which means a number of 2<sup>10</sup> = 1024 rounds.
```
Comment (atennapel) [inline content/refguide/project-settings.md:108]: I wonder if so much detail is needed. The fact that it uses powers of 2 might not be so relevant.

Comment (jveldhuizen) [reply content/refguide/project-settings.md:108]: Do you think the next sentence is good enough:

> This value is used to specify the cost for the BCrypt algorithm. This is 10 by default.

Comment (atennapel) [reply content/refguide/project-settings.md:108]: Yes that's good.

**Comment by atennapel** [comment]
Looks good to me.

**Comment by Adam-Dupaski** [comment]
@NatasaKralj this needs review and verification for which version it will appear in. When can you prepare it?

**Comment by Adam-Dupaski** [comment]
@NatasaKralj it is probably for 9.11 - see RN.

**Comment by Adam-Dupaski** [comment]
@NatasaKralj please verify if same change will be made for Mx8 doc to correspond to RN in #4198

**Comment by NatasaKralj** [comment]
@Adam-Dupaski will start this today, should be done by end of week.

File: content/refguide/project-settings.md [hunk 2]
```diff
@@ -103,15 +103,19 @@ The hash algorithm is used to generate hash values for attributes of the **Hashe
...
 
-Imagine an empty microflow that returns nothing at all exposed as a published web service. We ask one user to execute this operation as many times as he can in one minute (simulated with SoapUI). First we set the hashing algorithm to BCrypt, then we set it to SSHA256. Any extra overhead here (on top of establishing the connection, building the XML message and so forth) is basically the hashing algorithm, as the operation should take near zero milliseconds and there is no result. So that leaves only the login, or, more precisely, the hashing of the password.
+Imagine an empty microflow that returns nothing at all exposed as a published web service. We ask one user to execute this operation as many times as he can in one minute (simulated with SoapUI). First we set the hashing algorithm to BCrypt (with cost value 10), then we set it to SSHA256. Any extra overhead here (on top of establishing the connection, building the XML message and so forth) is basically the hashing algorithm, as the operation should take near zero milliseconds and there is no result. So that leaves only the login, or, more precisely, the hashing of the password.
```
Comment (Adam-Dupaski) [inline content/refguide/project-settings.md:118]: @NatasaKralj please look at any formatting needed for "SSHA256".

File: content/refguide/project-settings.md [hunk 3]
```diff
@@ -131,7 +135,7 @@ So 80 milliseconds per operation is not that much, right? Well, that depends on
 The difference is noticeable when the operation takes less time. So if you expect a very high amount of concurrency in operations where hashing takes place (most commonly any place where login operations are involved), you might want to consider changing your hashing algorithm.
 
 {{% alert type="info" %}}
-It is important to remember when changing hashing algorithms is that any hashed attribute (like the System$User password attribute) has its algorithm set on hashing. In other words, for the hashing type to take effect, any existing hashed attribute will have to be reset using the new hashing type.
+It is important to remember when changing hashing algorithms that any hashed attribute (like the System$User password attribute) has its algorithm set on hashing. In other words, for the hashing type to take effect, any existing hashed attribute will have to be reset using the new hashing type.
```
Comment (Adam-Dupaski) [inline content/refguide/project-settings.md:138]: @NatasaKralj please look at any formatting needed for "System$User".

**Comment by NatasaKralj** [comment]
@atennapel Can you please confirm which versions of Studio Pro this change is for? 
Is it for Studio Pro 9.11.0, as suggested in the [RNs](https://github.com/mendix/docs/pull/4191)?
Is it for also Studio Pro 8.16.16, as suggested in the [RNs](https://github.com/mendix/docs/pull/4198)? 
If it's also for SP 8.16.16, refguide8 needs the same updates.

File: content/refguide/project-settings.md [hunk 4]
```diff
@@ -98,20 +98,24 @@ The hash algorithm is used to generate hash values for attributes of the **Hashe
...
+Mendix believes both algorithms are secure enough to store passwords within Mendix. The main difference between **BCrypt** and **SSHA256** is that the BCrypt algorithm has been configured so that it is relatively slow on purpose, since it was designed specifically to stop brute force attacks. That's why this results in a slight performance difference with the SSHA256 algorithm.
 
-#### 3.9.1 Performance
+#### 3.9.1 BCrypt Cost {#bcrypt-cost}
 
-This performance difference is hardly noticeable to a single user when signing in (the password you enter when signing in is hashed using the selected algorithm), so in general the performance alone is not a reason to choose SSHA256 over BCrypt. This situation can change when dealing with high concurrency of hashing operations. A common example of an area where this occurs is published web services exposing operations that compute quickly, like short-running microflows.
+The BCrypt cost value is used to specify the cost of the BCrypt algorithm. The default value is 10, and can go up to 30. The higher the value is, the slower the process of hashing values. For more information, see the subsections below.
```
[the remark this answers is not in the document]
Comment (atennapel) [reply content/refguide/project-settings.md:108]: It's a numeric up-down field in the Project Settings.

File: content/refguide/project-settings.md (hunk 2 above)
Comment (atennapel) [reply content/refguide/project-settings.md]: Indeed, they are contained in a drop down list.

File: content/refguide/project-settings.md (hunk 3 above)
Comment (NatasaKralj) [reply content/refguide/project-settings.md]: Formatted to code

File: content/refguide/project-settings.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (NatasaKralj) [reply content/refguide/project-settings.md]: I adjusted this to have the BCrypt cost bold, as it shows up in the UI.

**Review by NatasaKralj — APPROVED** [review]
Reviewed, updated. All looks good now.

**Pull request merged**